ufl/core/external_operator.py

```diff
@@ -0,0 +1,278 @@
+                error("Expecting a size of %s for %s" % (len(self.ufl_operands), derivatives))
+
+            self.derivatives = derivatives
+            # If we have arguments, the appropriate function space has already been set up upstream
+            if len(arguments) == 0:
+                # In this block, we construct the function space on which lives the ExternalOperator
+                # accordingly to the derivatives taken, for instance in 2d:
+                #      for V = VectorFunctionSpace(...)
+                #      if u = Function(V); e = ExternalOperator(u, function_space=V)
+                #      then, e.ufl_shape = (2,)
+                #            dedu.ufl_shape = (2,2)
+                #            ...
+                # Therefore, for 'dedu' we need to construct the new TensorElement to end up with the appropriate
+                # function space and we also need to store the VectorElement corresponding to V since it is on this
+                # function_space that we will interpolate the operands.
+                s = ref_coefficient.ufl_shape
+                for i, e in enumerate(self.derivatives):
+                    s += self.ufl_operands[i].ufl_shape * e
+                original_function_space = self._make_function_space(s, sub_element=ref_coefficient.ufl_element())
+                ufl_shape = original_function_space.ufl_element().reference_value_shape()
+            else:
+                ufl_shape = ref_coefficient.ufl_element().reference_value_shape()
+                original_function_space = ref_coefficient.ufl_function_space()
+        else:
+            ufl_shape = ref_coefficient.ufl_element().reference_value_shape()
+            original_function_space = ref_coefficient.ufl_function_space()
+            self.derivatives = (0,) * len(self.ufl_operands)
+
+        if coefficient is None:
+            coefficient = Coefficient(original_function_space, count=count)
+        elif not isinstance(coefficient, Coefficient):
+            raise TypeError('Expecting a Coefficient and not %s', type(coefficient))
+        self._coefficient = coefficient
+        self.ufl_shape = ufl_shape
+        self._original_function_space = self.coefficient.ufl_function_space()
+
+        self._extop_dependencies = [self, ]
+
+        if self.derivatives == (0,) * len(self.ufl_operands):
+            self._extop_master = self
+
+    @property
+    def coefficient(self):
+        return self._coefficient
+
+    def _extract_coeffs_and_args(self, operands):
+        from ufl.algorithms.analysis import extract_arguments
+        args = tuple((e, is_adjoint) for e, is_adjoint in operands if len(extract_arguments(e)) != 0)
+        ops = tuple((e, is_adjoint) for e, is_adjoint in operands if (e, is_adjoint) not in args)
+        return ops, args
+
+    def arguments(self):
+        return self._arguments
+
+    def action_args(self):
+        return self._action_args
+
+    @property
+    def _extop_items(self):
```

Comment:
This is a tremendously non-intuitive name.

Reply:
Yes indeed, I was not really imaginative when I wrote this. I had to remove this anyway as it only appears once (in `ufl/algorithms/analysis`)